proto/api/v1/event_service.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+option go_package = "v1";
+
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/common.proto";
+import "api/v1/pagination.proto";
+import "google/protobuf/timestamp.proto";
+
+package v1;
+
+message EventsResponse {
+    repeated Event events      = 1;
+    Pagination     pagination  = 2;
+}
+
+message GetEventsRequest {
+    google.protobuf.Timestamp from      = 1;
+    map<string, string> labels_to_match = 2;
+}
+
+// Events are emitted by Central. They are used to create transparency for users
+// for asynchronous, background tasks.
+// Currently, events only cover logs emitted by Central.
+message Event {
+    // UUID of the event.
+    string id                            = 1;
+
+    // Message associated with the event. The message may include detailed information
+    // as well as potential actions that can be taken for this particular event.
+    string message                       = 2;
```

Comment:
I'm wondering if a single string message is a good way to represent events. I'm concerned that it doesn't map well to structured logs, and eventually we'll be parsing json in message. Maybe we could use https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto#L51 to represent structured content?

Reply:
That's what the `labels` field would be for.

The `message` field here would be the `msg` field within a structured log message. Other additional fields would be added under labels. That keeps it generic enough to allow also events emitted from things other than logs to fill the message field without workarounds (in your case of the struct, how would it look like for e.g. the UI to retrieve the message to display, and what would be the diff of the struct field compared to the `labels` field?).

Reply:
The difference to `labels` would be that `Struct` can be nested and typed. For example it could be useful if some log context has an integer field, and in the future we would like to add advanced search queries like `integer_field > X`. But I guess with protobuf we are relatively safe and could always add a `Struct` field later on if we want to support more advanced structured data.

Reply:
Yeah, fair point that does make sense for more advanced queries. I suppose we can do this later then.